### PR TITLE
[ExportVerilog] Format specifiers: hierpath separation character

### PR DESCRIPTION
**Pull request by teqdruid**
Support format specifiers on Verbatim*Op substitutions. The first one only applies to hierpaths: the separation character.

### CODE REVIEW

File: docs/Dialects/SV/RationaleSV.md [hunk 1]
```diff
@@ -141,6 +141,21 @@ sv.verbatim "MACRO({{0}}, {{1}} reg={{4}}, {{3}})"
             {symRefs = [@reg1, @Module1, @instance1]}
 ```
 
+Substitions also allow format specifier after a ':'. The meaning of said options
+depends on the of the operand or symbol. So far, the follow format specifiers
```
Comment (dtzSiFive) [inline docs/Dialects/SV/RationaleSV.md:145]: Looks like missing a word, maybe? "on the of the operand or symbol" -- not quite sure.

Also:
```suggestion
depends on the of the operand or symbol. So far, the following format specifiers
```

File: docs/Dialects/SV/RationaleSV.md [hunk 2]
```diff
@@ -141,6 +141,21 @@ sv.verbatim "MACRO({{0}}, {{1}} reg={{4}}, {{3}})"
...
+are supported:
+
+- Symbol refering to a `hw.hierpath`: the separation character for joining names
+in the path. Defaults to '.'.
```
Comment (seldridge) [inline docs/Dialects/SV/RationaleSV.md:149]: As written in `ExportVerilog`, I think this is a "separation string". If this is intended, then the documentation should talk about it as such as opposed to a separation "character". (Also, is the separation string behavior intended?)

**Review by seldridge — APPROVED** [review]
Generally looks good. Question about "separator string" vs. "separator character". I'm not sure it matters other than it's consistent. E.g., it may be useful to have a separator string of `", "` which will let you mimic string interleaving behavior.

**Comment by teqdruid** [comment]
@seldridge I meant separator string, rather than character.

File: lib/Conversion/ExportVerilog/ExportVerilog.cpp [hunk 3]
```diff
@@ -1226,6 +1226,16 @@ void EmitterBase::emitTextWithSubstitutions(
...
         continue;
       }
+      size_t operandNoLength = next - start;
+
+      // Format string options follow a ':'.
+      StringRef fmtOptsStr;
+      if (string[next] == ':') {
+        size_t startFmtOpts = next + 1;
+        while (next < string.size() && string[next] != '}')
+          ++next;
```
[the remark this answers is not in the document]
Comment (teqdruid) [reply lib/Conversion/ExportVerilog/ExportVerilog.cpp:1236]: Yes. The whole string makes up the specifier which would be interpreted by the individual types. Plus, I meant string.

**Pull request merged**